### PR TITLE
add API endpoints for getting a random legal deck

### PR DESCRIPTION
**Pull request by adambiltcliffe**
/api/randomlegaldeckurl/ returns a JSON object with a string property 'url' that gives the page for a random deck from the current season. For use in #5657

### CODE REVIEW

File: decksite/api.py [hunk 1]
```diff
@@ -18,6 +18,20 @@ def deck_api(deck_id):
     blob = deck.load_deck(deck_id)
     return return_json(blob)
 
+@APP.route('/api/randomlegaldeck/')
```
Comment (silasary) [inline decksite/api.py]: No trailing slash

File: decksite/api.py [hunk 2]
```diff
@@ -18,6 +18,20 @@ def deck_api(deck_id):
...
+def random_deck_api():
+    blob = deck.random_legal_deck()
+    return return_json(blob)
+
+@APP.route('/api/randomlegaldeckurl/')
```
Comment (silasary) [inline decksite/api.py]: This seems unneccessary

Comment (silasary) [reply decksite/api.py]: Just use the above endpoint and extract the url for it

**Review by silasary — CHANGES_REQUESTED** [review]

File: decksite/api.py (hunk 1 above)
Comment (adambiltcliffe) [reply decksite/api.py]: Removed

File: decksite/api.py (hunk 2 above)
Comment (adambiltcliffe) [reply decksite/api.py]: Removed /api/randomlegaldeckurl and added url attribute to the returned data from /api/randomlegaldeck

**Review by silasary — APPROVED** [review]

**Pull request merged**